File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F19.P1.go
package main

import "fmt"

func concurrentSum() int {
	// BEGIN PROBE F19.P1
	ch := make(chan int)
	go func() { ch <- 20 }()
	go func() { ch <- 22 }()
	sum := <-ch + <-ch
	return sum
	// END PROBE F19.P1
}

func main() {
	fmt.Println(concurrentSum())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F19.P1.cpp
#include <cstdint>
#include <cstdio>
#include <future>

std::int32_t concurrent_sum() {
// BEGIN PROBE F19.P1
auto a = std::async(std::launch::async, []{ return 20; });
auto b = std::async(std::launch::async, []{ return 22; });
std::int32_t sum = a.get() + b.get();
return sum;
// END PROBE F19.P1
}

int main() {
    std::printf("%d\n", concurrent_sum());
}
